client/extensions/woocommerce/app/product-categories/index.js

```diff
@@ -1,29 +1,113 @@
 /**
  * External dependencies
  */
-
 import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
 import classNames from 'classnames';
 import { connect } from 'react-redux';
 import { localize } from 'i18n-calypso';
+import { union, includes, trim, debounce } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import ActionHeader from 'woocommerce/components/action-header';
+import { DEFAULT_PRODUCT_CATEGORIES_PER_PAGE } from 'woocommerce/state/sites/product-categories/utils';
+import { fetchProductCategories } from 'woocommerce/state/sites/product-categories/actions';
 import { getLink } from 'woocommerce/lib/nav-utils';
 import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
 import Main from 'components/main';
 import NavTabs from 'components/section-nav/tabs';
 import NavItem from 'components/section-nav/item';
+import ProductCategoriesList from 'woocommerce/app/product-categories/list';
 import SectionNav from 'components/section-nav';
 import Search from 'components/search';
 
 class ProductCategories extends Component {
 
+	state = {
+		requestedPages: [ 1 ],
+		requestedSearchPages: [],
+	};
+
+	constructor( props ) {
+		super( props );
+		this.debouncedOnSearch = debounce( this.onSearch, 500 );
+	}
+
+	componentWillMount() {
+		const { site } = this.props;
+		this.props.fetchProductCategories( site.ID, {
+			page: 1,
+			per_page: DEFAULT_PRODUCT_CATEGORIES_PER_PAGE,
```

Comment:
Should we use `DEFAULT_PRODUCT_CATEGORIES_PER_PAGE` in the default query in [here](https://github.com/Automattic/wp-calypso/blob/ed678435106f85506e1f11b4597ed549584be417/client/extensions/woocommerce/state/sites/product-categories/utils.js#L14) so we don't have to pass in the `per_page` here? Guessing that could result in a performance hit on the edit products pages requesting 100 vs 10 there too.